packages/core/src/__stampers__/api/web/stamper.ts

```diff
@@ -0,0 +1,238 @@
+      request.onsuccess = () => {
+        db.close();
+        resolve(request.result as string[]);
+      };
+      request.onerror = () => {
+        db.close();
+        reject(request.error);
+      };
+    });
+  }
+
+  async createKeyPair(externalKeyPair?: CryptoKeyPair): Promise<string> {
+    let privateKey: CryptoKey;
+    let publicKey: CryptoKey;
+    if (externalKeyPair) {
+      const extractable = (externalKeyPair.privateKey as any).extractable;
+      if (extractable !== false) {
+        throw new Error("Provided privateKey must be non-extractable.");
+      }
+      privateKey = externalKeyPair.privateKey;
+      publicKey = externalKeyPair.publicKey;
+    } else {
+      const keyPair = await crypto.subtle.generateKey(
+        { name: "ECDSA", namedCurve: "P-256" },
+        false, // Non-extractable private key
+        ["sign", "verify"],
+      );
+      privateKey = keyPair.privateKey;
+      publicKey = keyPair.publicKey;
+    }
+    const rawPubKey = new Uint8Array(
+      await crypto.subtle.exportKey("raw", publicKey),
+    );
+    const compressedPubKey = pointEncode(rawPubKey);
+    const compressedHex = uint8ArrayToHexString(compressedPubKey);
+    await this.storeKeyPair(compressedHex, privateKey);
+    return compressedHex;
+  }
+
+  async deleteKeyPair(publicKeyHex: string): Promise<void> {
+    const db = await this.openDb();
+    return new Promise((resolve, reject) => {
+      const tx = db.transaction(DB_STORE, "readwrite");
+      const store = tx.objectStore(DB_STORE);
+      store.delete(publicKeyHex);
+      tx.oncomplete = () => {
+        db.close();
+        resolve();
+      };
+      tx.onerror = () => reject(tx.error);
+    });
+  }
+
+  async clearKeyPairs(): Promise<void> {
+    const db = await this.openDb();
+    return new Promise((resolve, reject) => {
+      const tx = db.transaction(DB_STORE, "readwrite");
+      const store = tx.objectStore(DB_STORE);
+      store.clear();
+      tx.oncomplete = () => {
+        db.close();
+        resolve();
+      };
+      tx.onerror = () => reject(tx.error);
+    });
+  }
+
+  private async sign(payload: string, publicKeyHex: string): Promise<string> {
+    const privateKey = await this.getPrivateKey(publicKeyHex);
+    if (!privateKey) {
+      throw new Error("Key not found for publicKey: " + publicKeyHex);
+    }
+    const encodedPayload = new TextEncoder().encode(payload);
+    const signatureIeee1363 = await crypto.subtle.sign(
+      { name: "ECDSA", hash: { name: "SHA-256" } },
+      privateKey,
+      encodedPayload,
+    );
+    const signatureDer = convertEcdsaIeee1363ToDer(
+      new Uint8Array(signatureIeee1363),
+    );
+    return uint8ArrayToHexString(signatureDer);
+  }
+
+  async stamp(payload: string, publicKeyHex: string): Promise<TStamp> {
+    const signature = await this.sign(payload, publicKeyHex);
+    const stamp = {
+      publicKey: publicKeyHex,
+      scheme: "SIGNATURE_SCHEME_TK_API_P256",
```

Comment:
I feel like this should be brought up as a top-level comment: this indexedDB stamper manages P256 key pairs only.